test/e2e/framework/framework.go

```diff
@@ -306,26 +308,35 @@ func (f *Framework) CreateServiceAccount(namespace, serviceAccount string) (Clea
 }
 
 func (f *Framework) GetServiceAccountToken(namespace, name string) (string, error) {
-	ctx := context.Background()
-	var token string
-	err := Poll(5*time.Second, time.Minute, func() error {
-		secrets, err := f.KubeClient.CoreV1().Secrets(namespace).List(ctx, metav1.ListOptions{})
+	var (
+		ctx             = context.Background()
+		token           string
+		tokenExpiration = time.Duration(time.Hour * 12)
+		expirationTime  = metav1.NewTime(time.Now().Add(tokenExpiration))
+	)
+	err := Poll(time.Second, time.Minute, func() error {
```

Comment:
Just curious about the change in the poll interval, any reason about the change in the poll interval? that's just a nit but I'm wondering if this would be more poll intensive.

Reply:
The previous approach was to extract the token from the SA secret and it could take time from Kubernetes to populate the secret. The new approach gets a token from the API which should succeed from the first try in the vast majority of runs.

Reply:
Thanks for the clarifications ;)